src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -28,8 +28,17 @@
 namespace chip {
 namespace scenes {
 
-static constexpr uint16_t kMaxScenesPerFabric   = (MATTER_SCENES_TABLE_SIZE - 1) / 2;
+#ifdef MATTER_SCENES_TABLE_SIZE
 static constexpr uint16_t kMaxScenesPerEndpoint = MATTER_SCENES_TABLE_SIZE;
+#else
+static constexpr uint16_t kMaxScenesPerEndpoint = CHIP_CONFIG_MAX_SCENES_TABLE_SIZE;
+#endif
+
+static_assert(kMaxScenesPerEndpoint <= CHIP_CONFIG_MAX_SCENES_TABLE_SIZE,
+              "CHIP_CONFIG_MAX_SCENES_TABLE_SIZE is smaller than the zap configuration, please increase "
+              "CHIP_CONFIG_MAX_SCENES_TABLE_SIZE in CHIPConfig.h if you really need more scenes");
+static_assert(kMaxScenesPerEndpoint >= 16, "Per spec, kMaxScenesPerEndpoint must be greater than 16");
```

Comment:
There's a mismatch between the description in the string literal and the actual static assertion condition. The description suggests that kMaxScenesPerEndpoint should be greater than 16, while the condition checks that kMaxScenesPerEndpoint is greater than or equal to 16.

To align the description with the check, the assertion message should be modified to:
static_assert(kMaxScenesPerEndpoint >= 16, "Per spec, kMaxScenesPerEndpoint mu

Reply:
Changed to " must be at least 16"